pkg/wharfapi/token.go

```diff
@@ -67,15 +68,40 @@ func (c Client) GetToken(token string, userName string) (Token, error) {
 	return tokens[0], nil
 }
 
+func (c Client) PutToken(token Token) (Token, error) {
+	body, err := json.Marshal(token)
+	if err != nil {
+		return Token{}, err
+	}
+
+	log.WithField("project", string(body)).Traceln()
+
+	apiURL := fmt.Sprintf("%s/api/token", c.APIURL)
+	ioBody, err := doRequest("PUT | TOKEN |", http.MethodPut, apiURL, body, c.AuthHeader)
+	if err != nil {
+		return Token{}, err
+	}
+
+	defer (*ioBody).Close()
+
+	newToken := Token{}
+	err = json.NewDecoder(*ioBody).Decode(&newToken)
+	if err != nil {
+		return Token{}, err
+	}
```

Comment:
nit: Same as other comment; this can be inlined

```suggestion
	if err := json.NewDecoder(*ioBody).Decode(&newToken); err != nil {
		return Token{}, err
	}
```